edb/server/compiler_pool/pool.py

```diff
@@ -618,6 +625,24 @@ def get_debug_info(self):
     def get_size_hint(self) -> int:
         raise NotImplementedError
 
+    def _maybe_update_last_active_time(self):
+        if sys.exc_info()[0] is None:
+            self._last_active_time = time.monotonic()
+
+    async def health_check(self) -> bool:
+        elapsed = time.monotonic() - self._last_active_time
+        if elapsed > HEALTH_CHECK_MIN_INTERVAL:
+            task = self._loop.create_task(
+                self.make_compilation_config_serializer()
+            )
+            task.add_done_callback(
+                lambda _: self._maybe_update_last_active_time()
+            )
+            await asyncio.wait([task], timeout=HEALTH_CHECK_TIMEOUT)
+            if not task.done():
+                return False
```

Comment:
[Unlike wait_for(), wait() does not cancel the futures when a timeout occurs.](https://docs.python.org/3/library/asyncio-task.html#waiting-primitives)

so we should cancel them on timeout to avoid leaking that memory and clogging up the event loop

Reply:
It was intentional (to avoid hairy asyncio cancellation, should've added a gotcha), but okay, will fix

> to avoid leaking that memory and clogging up the event loop

to avoid adding more burden to the already overwhelmed compiler processes. FYI we don't yet have a way to cancel an RPC that is already sent to a compiler process, the cancellation here is only meaningful while the RPC is in the queue, which is ~90% of the case I think.